tasks/agent.py

```diff
@@ -319,12 +321,33 @@ def omnibus_build(ctx, puppy=False, log_level="info", base_dir=None, gem_path=No
             "overrides": overrides_cmd,
             "populate_s3_cache": ""
         }
+        pfxfile = None
+        if sys.platform == 'win32' and os.environ.get('SIGN_WINDOWS'):
+            # get certificate and password from ssm
+            pfxfile = get_signing_cert(ctx)
+            pfxpass = get_pfx_pass(ctx)
+            print("Retrieved file {} pass {}".format(pfxfile, pfxpass))
+            # hack for now.  Remove `sign_windows, and set sign_pfx`
+            env['SIGN_PFX'] = "{}".format(pfxfile)
+            env['SIGN_PFX_PW'] = "{}".format(pfxpass)
+
         if omnibus_s3_cache:
             args['populate_s3_cache'] = " --populate-s3-cache "
         if skip_sign:
             env['SKIP_SIGN_MAC'] = 'true'
+
         env['PACKAGE_VERSION'] = get_version(ctx, include_git=True, url_safe=True)
-        ctx.run(cmd.format(**args), env=env)
+
+        try:
+            ctx.run(cmd.format(**args), env=env)
+        except Exception as e:
+            if pfxfile:
+                os.remove(pfxfile)
+            raise
+
+        if pfxfile:
+            os.remove(pfxfile)
```

Comment:
I do not know how sensitive is this file (It might because of `mkstemp`) but if you really want to remove it, you can start the `try` at line 325
```python
try:
   if sys.platform == 'win32' and os.environ.get('SIGN_WINDOWS'): // Line 325
   ....
  ctx.run(cmd.format(**args), env=env) // line 342
finally:
   if pfxfile:
     os.remove(pfxfile)
```